x-pack/platform/plugins/shared/cases/public/components/markdown_editor/paste_upload/use_file_upload_api.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect } from 'react';
+import type { Dispatch } from 'react';
+import type { Subscription } from 'rxjs';
+import type { FileState, UploadState } from '@kbn/shared-ux-file-upload/src/upload_state';
+import { type PasteUploadState, UploadPhase, ActionType, type Action } from './types';
+
+const canUpload = (files: FileState[], uiState: PasteUploadState) => {
```

Comment:
Should this also include a mime type compatibility check? Or is that being handled somewhere else? Or should we handle it in both places?

Reply:
We perform this check in the [initial event handler](https://github.com/elastic/kibana/pull/226077/files#diff-6cd863e3197f596430864c5d6e37a65271c3cd89a3da045e80bd57402dabac59R124) when the paste interaction occurs.